src/main/native/com/code_intelligence/jazzer/driver/android_tooling.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dlfcn.h>
+#include <jni.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include "com_code_intelligence_jazzer_android_AndroidRuntime.h"
+
+const char *RUNTIME_LIBRARY = "libandroid_runtime.so";
+const char *EMBEDDED_DIR = "/data/fuzz:/system/lib64";
+const char *USE_EMBEDDED = "use_embedded";
+
+std::string getNativeString(JNIEnv *env, jbyteArray javaString) {
+  jsize num_bytes = env->GetArrayLength(javaString);
+  jbyte *elements = env->GetByteArrayElements(javaString, NULL);
+
+  if (!elements) {
+    std::cerr << "ERROR: Passed a invalid runtime_libs string." << std::endl;
+    exit(1);
+  }
+  std::string nativeString(reinterpret_cast<char *>(elements), num_bytes);
+
+  // Release the element array provided by JNI
+  env->ReleaseByteArrayElements(javaString, elements, JNI_ABORT);
+
+  return nativeString;
+}
+
+// Register native methods from the Android Runtime (ART) framework.
+[[maybe_unused]] jint
+Java_com_code_1intelligence_jazzer_android_AndroidRuntime_registerNatives(
+    JNIEnv *env, jclass clazz, jbyteArray runtimeLibs) {
```

Comment:
nit: This param name and description in Opt.java is confusing to me. 

runtimeLibs and "Which libraries to use when initializing ART (native launcher only)")

to me means a list of libraries to use when init-ing ART. It would be nice to find a better name:

initialization_options="use_embedded"

or something similar

Reply:
Updated the name of the param.  Also updated the name of the param in the ingestion pipeline, added you to a separate CL to approve.